pyomo/common/tests/test_tee.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright 2017 National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+import os
+import time
+
+from six import StringIO, BytesIO
+
+from pyomo.common.log import LoggingIntercept
+import pyomo.common.unittest as unittest
+
+import pyomo.common.tee as tee
+
+class TestTeeStream(unittest.TestCase):
+    def test_stdout(self):
+        a = StringIO()
+        b = StringIO()
+        with tee.TeeStream(a,b) as t:
+            t.STDOUT.write("Hello\n")
+        self.assertEqual(a.getvalue(), "Hello\n")
+        self.assertEqual(b.getvalue(), "Hello\n")
+
+    def test_err_and_out_are_different(self):
+        with tee.TeeStream() as t:
+            out = t.STDOUT
+            self.assertIs(out, t.STDOUT)
+            err = t.STDERR
+            self.assertIs(err, t.STDERR)
+            self.assertIsNot(out, err)
+
+    @unittest.skipIf(not tee._peek_available,
+                     "This test is not deterministic if select/peek "
+                     "are available")
+    def test_merge_out_and_err(self):
+        a = StringIO()
+        b = StringIO()
+        with tee.TeeStream(a,b) as t:
+            # This is a slightly nondeterministic (on Windows), so a
+            # flush() and short pause should help
+            t.STDOUT.write("Hello\nWorld")
+            t.STDOUT.flush()
+            time.sleep(tee._poll_interval*1.1)
+            t.STDERR.write("interrupting\ncow")
+            t.STDERR.flush()
+            time.sleep(tee._poll_interval*2)
+        self.assertEqual(a.getvalue(), "Hello\ninterrupting\ncowWorld")
+        self.assertEqual(b.getvalue(), "Hello\ninterrupting\ncowWorld")
+
+    def test_merged_out_and_err_without_peek(self):
+        a = StringIO()
+        b = StringIO()
+        try:
+            _tmp, tee._peek_available = tee._peek_available, False
+            with tee.TeeStream(a,b) as t:
+                # Ensure both threads are running
+                t.STDOUT
+                t.STDERR
+                # ERR should come out before OUT, but this is slightly
+                # nondeterministic, so a short pause should help
```

Comment:
Given this comment should this test have the same skipif condition as the previous test?

Reply:
No.  The previous comment was unclear (the determinism wasn't really the issue: the previous test tested the expected behavior of the `_mergedReader` (which requires `_peek_available`),  This test should always work.  Both are slightly nondeterministic, but the delays are hopefully long enough to prevent false failures.